src/apply.c

```diff
@@ -650,7 +649,12 @@ int git_apply_to_tree(
 	 */
 	for (i = 0; i < git_diff_num_deltas(diff); i++) {
 		delta = git_diff_get_delta(diff, i);
-
+		if (delta->status == GIT_DELTA_ADDED &&
+			git_index_find(NULL, postimage, delta->new_file.path) != GIT_ENOTFOUND) {
+				git_error_clear();
+				error = GIT_EEXISTS;
```

Comment:
I think that we should check the return code more distinctly - we should only return `GIT_EEXISTS` when `git_index_find` returns 0, otherwise we should propagate its return information.

Reply:
After testing out this commit, I think this logic is incorrect. This does detect if the diff had a duplicate file, but it also returns `GIT_EEXISTS` for valid index+diff applications.

The postimage is initialized as a copy of the preimage. So if the preimage already contains file `foo`, and the diff also attempts to add file `foo`, it will return `GIT_EEXISTS` here. But we should actually produce an index with conflicts in that case, rather than returning an error.